ci/docker/ubuntu-24.04-cpp.dockerfile

```diff
@@ -141,9 +141,14 @@ RUN if [ "${gcc_version}" = "" ]; then \
           add-apt-repository ppa:ubuntu-toolchain-r/volatile; \
```

Comment:
Can we use `ppa:ubuntu-toolchain-r/ppa` ( https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/ppa ) instead?
It seems that this doesn't have gcc-14 nor gcc-15: https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/volatile

https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/volatile has gcc-15 but it's for Ubuntu 25.04.

It may be better that we use Debian GNU/Linux experimental instead: https://packages.debian.org/search?keywords=gcc-15
If Debian GNU/Linux experimental is OK, I can work on it.